internal/util/resource/resource.go

```diff
@@ -73,31 +73,26 @@ func Track(obj any, token *Token) {
 	}
 
 	// use token instead of obj itself,
-	// because otherwise profile will hold a reference to obj and finalizer will never run
+	// because otherwise profile will hold a reference to obj and cleanup will never run
 	p.Add(token, 1)
 
-	var stack string
+	var errMsg = fmt.Sprintf("%T has not been finalized", obj)
```

Comment:
Let's use the short form to be consistent with the rest of the code :hugs: 

```suggestion
	errMsg := fmt.Sprintf("%T has not been finalized", obj)
```

Reply:
Done :)